ruby/data.rb

```diff
@@ -111,14 +113,38 @@
 			# ... but the webserver user may not have the write permission to the ".." directory.
 			run "rmdir #{shared_path}/assets"
 
-			upload data_path, shared_path, :recursive => true, :via => :scp
+			source = data_path
+			target = find_servers_for_task(current_task)[0].host << ":" << shared_path << "/assets"
```

Comment:
Ummm not sure - I think the intention here was to transfer assets in parallel to all servers. If you upload to only one it could take long time for them to sync up between backends, which we want to avoid (csync I'm looking at you).